source/agora/consensus/protocol/Nominator.d

```diff
@@ -624,19 +859,86 @@ extern(D):
 
         // enrollment data may be empty, but not transaction set
         if (data.tx_set.length == 0)
-            assert(0, format!"Transaction set empty for slot %u"(slot_idx));
+            assert(0, format!"Transaction set empty for slot %s"(height));
 
-        log.info("Externalized consensus data set at {}: {}", slot_idx, data);
+        log.info("Externalized consensus data set at {}: {}", height, prettify(data));
         try
         {
-            if (!this.ledger.onExternalized(data))
-                assert(0);
+            auto blocks = this.ledger.getBlocksFrom(Height(slot_idx - 1));
+            assert(!blocks.empty);  // should not happen
+            const prev_block = blocks.front;
+
+            const block = makeNewBlock(prev_block, data.tx_set, data.enrolls);
+
+            const signed_block = updateMultiSignature(block);
+
+            // If we did not sign yet then add signature
+            if (this.schnorr_pair.V !in this.slot_sigs[height])
+            {
+                log.trace("ADD BLOCK SIG at height {} for this node {}", height, this.public_key);
+                this.slot_sigs[height][this.schnorr_pair.V] = createBlockSignature(block);
+            }
+            if (signed_block == Block.init)
+            {
+                // Do not externalize if block was not signed
```

Comment:
I think this should say "if we have not collected at least 50% + 1 signatures", right?

Reply:
Removed from this commit.